Automattic/wp-calypso
# SSR: Periodically snapshot cache keys to investigate spike in cache misses

Resurrecting #13419.

Use the logstash API endpoint to take a snapshot of SSR cache keys.

* Add data-layer for the endpoint
* For 0.1% of misses (we are seeing 9000 an hour) log: the key that misses, all existing keys in the cache, the user-agent, and the request path.


## Testing
* Checkout this branch
* build with `ENABLE_FEATURES=ssr/always-log-cache-misses npm start`
* hit a few /themes routes, like /themes/blog, /themes/music
* In Kibana, search on feature:calypso_ssr
* You should see cache miss logs, with the detail in the `extra` field:
<img width="1026" alt="screen shot 2018-03-12 at 17 35 34" src="https://user-images.githubusercontent.com/7767559/37299736-02a08af6-261c-11e8-8722-0c7d9952bb2c.png">
